curated_transformers/models/attention.py

```diff
@@ -44,15 +44,24 @@ def __init__(self, *, dropout_prob: float = 0.1):
         self.dropout = torch.nn.Dropout(p=dropout_prob)
 
     def forward(
-        self, k: Tensor, q: Tensor, v: Tensor, attn_mask: AttentionMask
+        self, k: Tensor, q: Tensor, v: Tensor, attn_mask: Optional[AttentionMask]
     ) -> Tensor:
         """
+        Apply attention layer to the given key, query and value.
+
+        Sequence elements that are marked with `False` in the attention mask
+        are ignored by the attention mechanism (if a mask is provided).
+
+        k (Tensor): key.
+        q (Tensor): query.
+        v (Tensor): value.
+        attn_mask (Optional[AttentionMask]): attention mask.
         Shapes:
             k, q, v - (batch, heads, seq_len, width)
             attn_mask - (batch, seq_len)
```

Comment:
I think it makes sense to also document the output shape (in all modules)?

Reply:
Done for the attention module.